nixos/modules/virtualisation/nixos-containers.nix

```
@@ -483,7 +483,7 @@ in
                       extraConfig = {
                         _file = "module at ${__curPos.file}:${toString __curPos.line}";
                         config = {
-                          nixpkgs = { inherit localSystem; };
+                          nixpkgs = { inherit hostPlatform; };
```

Maybe something like this, in order to support the old style as well.

```suggestion
                          nixpkgs =
                            if options.nixpkgs.hostPlatform.isDefined
                              then { inherit hostPlatform; }
                              else { inherit localSystem};
```